### PR TITLE
Fix #3406: [RTL] High-fi Recently- Played Activity List

### LINKED ISSUE #3406: [RTL] High-fi Recently- Played Activity List
Recently- Played Activity needs RTL support

screenshot

### PR DESCRIPTION
**Pull request by veena14cs**
## Explanation
This PR fixes #3406 . This PR fixes rtl support in `ongoingStoryCard`.

## Screenshot LTR and RTL
Mobile:
screenshot.......screenshot

screenshot....screenshot

Tablet:

screenshot....screenshot

screenshot......screenshot

## Tests with RTL code
Screenshot (53)

## Tests without RTL code fails as expected
Screenshot (54)

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/res/layout-land/ongoing_story_card.xml
Comment (rt4914) [inline app/src/main/res/layout-land/ongoing_story_card.xml:66]: Can you confirm these in mocks? Share link of the mock where it has been shown as `2 lines`.

File: app/src/main/res/layout-land/ongoing_story_card.xml
Comment (rt4914) [inline app/src/main/res/layout-land/ongoing_story_card.xml:68]: Also for this change we need mock link

File: app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingListAdapter.kt
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingListAdapter.kt:64]: Thanks for getting rid of this really complex code. :-D

**Review by rt4914 — CHANGES_REQUESTED** [review]
@veena14cs Great work. Suggest need mock links to verify the xml changes are correct or not.

File: app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingListAdapter.kt
Comment (veena14cs) [reply app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingListAdapter.kt:64]: :)

File: app/src/main/res/layout-land/ongoing_story_card.xml
Comment (veena14cs) [reply app/src/main/res/layout-land/ongoing_story_card.xml:66]: In the link https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/0e6ba726-1930-4f39-9c73-51df10a1c766/specs/ number of lines is not added in the mock. To keep it consistent with other versions of this xml files I changed it to `lines=2`

File: app/src/main/res/layout-land/ongoing_story_card.xml
Comment (veena14cs) [reply app/src/main/res/layout-land/ongoing_story_card.xml:68]: Text is bold in the mock link in the link https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/0e6ba726-1930-4f39-9c73-51df10a1c766/specs/ .

File: app/src/main/res/layout-land/ongoing_story_card.xml
Comment (rt4914) [reply app/src/main/res/layout-land/ongoing_story_card.xml:66]: Yes. It is 2 lines only.

File: app/src/main/res/layout-land/ongoing_story_card.xml
Comment (rt4914) [reply app/src/main/res/layout-land/ongoing_story_card.xml:68]: Thanks.

**Review by rt4914 — APPROVED** [review]
Nice work. Thanks

**Comment by veena14cs** [comment]
> Nice work. Thanks

Thanks @rt4914 :)

File: app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingListAdapter.kt
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingListAdapter.kt]: Nit: missing space between the brace & ``else``.

File: app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingStoryViewModel.kt
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingStoryViewModel.kt:50]: Can you write tests for this & the next function to make sure it computes the correct margin values for different positions?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @veena14cs! I think this great.

Would it be possible to add tests at the recently played activity level to verify that the different layouts presented in your screenshots are indeed laid out correctly in tests?

Edit: also, I think you're running into the same Bazel test failures here that are affecting your other PR.

**Comment by veena14cs** [comment]
> Edit: also, I think you're running into the same Bazel test failures here that are affecting your other PR.

Yes, as this also includes same change in MrginBindingAdapters. If the #3452 is fixed I think it will not have Bazel test failures.

File: app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingListAdapter.kt
Comment (veena14cs) [reply app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingListAdapter.kt]: Done.

**Comment by rt4914** [comment]
Un-assigning as we have discussed the solution in meeting.

**Comment by veena14cs** [comment]
> Un-assigning as we have discussed the solution in meeting.

Thanks I have updated testcases.

**Comment by veena14cs** [comment]
> Thanks @veena14cs! I think this great.
> 
> Would it be possible to add tests at the recently played activity level to verify that the different layouts presented in your screenshots are indeed laid out correctly in tests?
> 

@BenHenning Added tests PTAL.

File: app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingStoryViewModel.kt
Comment (veena14cs) [reply app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingStoryViewModel.kt:50]: Added tests for recyclerview item as it uses this function for setting start and end margins.

File: app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt]: We shouldn't use ktx synthetic references as they're now deprecated: https://developer.android.com/topic/libraries/view-binding/migration (though we don't use Jetpack view binding since we use AndroidX databinding, instead).

File: app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt]: Nit: let's order the tests consistently, e.g.:

- portrait
- landscape
- tabletPortrait
- tabletLandscape

File: app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt]:
```suggestion
    activity.supportFragmentManager
      .beginTransaction()
      .add(testFragment, TEST_FRAGMENT_TAG)
      .commitNow()
```

Wrapping should start at the first ``.`` starting the chain of operations.

File: app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt]: Nit: remove extra newline (only one is needed at the end of the file).

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @veena14cs. Left a couple of comments. Regarding the new tests & the shown failures, I only see 2 of the RTL tests failing in the screenshot. Shouldn't all of them fail if the new RTL code is reverted?

File: app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt
Comment (veena14cs) [reply app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt]: Done

File: app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt
Comment (veena14cs) [reply app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt]: Done.

**Comment by veena14cs** [comment]
> Thanks @veena14cs. Left a couple of comments. Regarding the new tests & the shown failures, I only see 2 of the RTL tests failing in the screenshot. Shouldn't all of them fail if the new RTL code is reverted?

Actually, the other 4 tests were added later and I had added 2 RTL tests earlier and its screenshot. I missed to update. I will do it now.

Edit: Updated screenshots with new tests.

File: app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt
Comment (veena14cs) [reply app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt]: It was unused import. Removed it.

**Comment by veena14cs** [comment]
> Un-assigning as we have discussed the solution in meeting.

Updated tests. PTAL

**Comment by BenHenning** [comment]
Thanks @veena14cs. I think this LGTM, but I can't confirm until #3452 is merged & this is updated to the latest develop. Please reassign to me at that point.

**Comment by veena14cs** [comment]
> Thanks @veena14cs. I think this LGTM, but I can't confirm until #3452 is merged & this is updated to the latest develop. Please reassign to me at that point.

Thanks Ben.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

File: app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingStoryViewModel.kt
Comment (anandwana001) [inline app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingStoryViewModel.kt:6]: with this addition, this file will go in the VIEW_MODEL_WTH_RESOURCES in app/BUILD.bazel file

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Will review again, once the blocking RP get's in.

**Comment by BenHenning** [comment]
Hi. Will need to look at this tomorrow.

Also, @veena14cs can you please fix the failing tests per CI?

**Comment by veena14cs** [comment]
I have merged with latest develop. This might fix CI.

**Review by BenHenning — APPROVED** [review]
Thanks @veena14cs. LGTM!

**Comment by BenHenning** [comment]
(also assigning back to @anandwana001, not sure if you meant to do that earlier)

File: app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingStoryViewModel.kt
Comment (veena14cs) [reply app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingStoryViewModel.kt:6]: Done.

**Comment by veena14cs** [comment]
> (also assigning back to @anandwana001, not sure if you meant to do that earlier)

Yes thanks.

**Review by anandwana001 — APPROVED** [review]
LGTM,
not the code owner for this PR, but keeping track of merging once all the checks pass.

**Pull request merged**
